src/common/sessionExportsManager.ts

```diff
@@ -0,0 +1,302 @@
+import z from "zod";
+import path from "path";
+import fs from "fs/promises";
+import { createWriteStream } from "fs";
+import { lock } from "proper-lockfile";
+import { FindCursor } from "mongodb";
+import { EJSON, EJSONOptions } from "bson";
+import { Transform } from "stream";
+import { pipeline } from "stream/promises";
+
+import { UserConfig } from "./config.js";
+import { Session } from "./session.js";
+import logger, { LogId } from "./logger.js";
+
+export const jsonExportFormat = z.enum(["relaxed", "canonical"]);
+export type JSONExportFormat = z.infer<typeof jsonExportFormat>;
+
+export type Export = {
+    name: string;
+    uri: string;
+    createdAt: number;
+};
+
+export type SessionExportsManagerConfig = Pick<
+    UserConfig,
+    "exportsPath" | "exportTimeoutMs" | "exportCleanupIntervalMs"
+>;
+
+const MAX_LOCK_RETRIES = 10;
+
+export class SessionExportsManager {
+    private availableExports: Export[] = [];
+    private exportsCleanupInterval: NodeJS.Timeout;
+    private exportsCleanupInProgress: boolean = false;
+
+    constructor(
+        private readonly session: Session,
+        private readonly config: SessionExportsManagerConfig
+    ) {
+        this.exportsCleanupInterval = setInterval(
+            () => void this.cleanupExpiredExports(),
+            this.config.exportCleanupIntervalMs
+        );
+    }
+
+    public async close() {
+        try {
+            clearInterval(this.exportsCleanupInterval);
+            const exportsDirectory = this.exportsDirectoryPath();
+            await fs.rm(exportsDirectory, { force: true, recursive: true });
+        } catch (error) {
+            logger.error(
+                LogId.exportCloseError,
+                "Error while closing SessionExportManager",
+                error instanceof Error ? error.message : String(error)
+            );
+        }
+    }
+
+    public exportNameToResourceURI(nameWithExtension: string): string {
+        this.validateExportName(nameWithExtension);
+        return `exported-data://${nameWithExtension}`;
```

Comment:
```suggestion
        return `exported-data://${encodeURIComponent(nameWithExtension)}`;
```

and the reverse when decoding?

Reply:
thank you for catching this 🙌, implemented this just now